design-documents/testing/functional/test-package.md

```diff
@@ -0,0 +1,96 @@
+
+![Packaging Process](img/packaging.png)
+
+2. Create `magento2-module` package as usual, but without tests.
+
+3. Out of excluded tests create a separate package.
+   - For this package we will automatically generate `composer.json` based on existing module `composer.json` file as shown on image below.
+    
+![Composer file](img/test_composer_json.png)
+
+### Version setter flow #1
+1. Versions will be set the same as for `magento2-module`. It means if we release `magento/module-store` version `101.0.3` we will specify same version for `magento/test-module-store`.
+2. Metapackage `magento/product-tests-community-edition` and `magento/project-tests-community-edition` created and have the same versions as Magento metapackage. 
+
+#### Benefits:
+1. Very easy to understand what version of tests you need to run against existing Magento instance
+2. Publication build will have small changes to enable this approach.
+
+#### Disadvantages: 
+1. Can not make separate releases for module or test. Always 2 packages will be released.
+
+### Version setter flow #2
+1. Versions will be set independently from Magento module based on SVC tool results.
+2. Metapackage `magento/product-tests-community-edition` and `magento/project-tests-community-edition` created and have the same versions as Magento metapackage.
+
+#### Benefits:
+1. Possibility to release test package only
+
+#### Disadvantages: 
+1. Bigger changes to Publication build
+    - require running SVC against latest release and current changes to determine package version to be set
+2. Not possible to make Backward incompatible changes without generating new Metapackage
+3. Difficult to determine what version of test package we should use against existing Magento Instance
+    - Open question about "What package requires what?" see in "[Open Questions](#Open-Questions)" block
+
+### Version setter flow #3
+1. Versions will be set independently from Magento module based on SVC tool results.
+2. Metapackage `magento/product-tests-community-edition` and `magento/project-tests-community-edition` created and have own versions based on smaller package version bumps.
+
+#### Benefits:
+1. Possibility to release test package only.
+2. Full independence in releases from Magento release lines.
+
+#### Disadvantages: 
+1. Bigger changes to Publication build
+   - require running SVC against latest release and current changes to determine package version to be set
+2. Not possible to make Backward incompatible changes without generating new Metapackage
+3. Difficult to determine what version of test package we should use against existing Magento Instance
+   - Open question about "What package requires what?" see in "[Open Questions](#Open-Questions)" block
+
+
+## Open Questions:
+1. What package requires what?
+   - Does Magento require-dev test packages?
+   - Does test packages require Magento module?
+   - Other potentials?
```

Comment:
I would say tests must not require magento modules because they can be installed remotely. 

Magento modules should also not depend on tests. Maybe tests can be added to `suggest` section.

It looks like we need some tool which will identify which test modules needed for specific Magento instance (via API calls).

Reply:
Do we want to support installing tests separately? Might be hard to check version compatibility if you want to install only some test modules but not all.

Reply:
Discussed in person and agreed that it might be better not to have dependency in test modules on Magento modules because of two reasons: 1) it would allow to install test modules separately from Magento, 2) we will not have to deal with issue of updating of Magento modules when installing test modules on Magento instance.

@okolesnyk could you investigate tooling that would allow to make sure that test modules compatible with Magento modules? One idea we had is to have dependency on Magento module specified in the extra section of composer.sjon of a test module. Tool that would collect list of installed modules on Magento instance and search for test packages in configured repositories and create composer.json with found test modules.